### PR TITLE
docs: Design proposal for Converged GTP tunnel setup

### PR DESCRIPTION
**Pull request by prabinakpattnaik**
Signed-off-by: prabinakpattnaik <prabinak@wavelabs.ai>

## Summary

Design proposal for add support for to make GTP tunnel operations by Sessiond for 4G Call as similar to 5G Call Flows.

### CODE REVIEW

File: docs/readmes/proposals/p025_Converged_GTP_Tunnel_Setup.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+The following tests scenario are proposed to verify to cover this functionality:
+
+### Integration Testing
+
+1. Verify the functionality using S1ap test cases.
+2. Write UT test cases for sessiond and pipelined.
+3. Write the python stub for verify the functionality.  
+
+### Regration Testing
```
Comment (electronjoe) [inline docs/readmes/proposals/p025_Converged_GTP_Tunnel_Setup.md]: I believe you meant `Regression Testing`?

**Review by electronjoe — CHANGES_REQUESTED** [review]

**Comment by electronjoe** [comment]
Please link to your GH Issue for this proposal in your PR description.

That and one nit typo - otherwise LGTM for review of the design over on the GH Issue.

File: docs/readmes/proposals/p025_Converged_GTP_Tunnel_Setup.md (hunk 1 above)
Comment (prabinakpattnaik) [reply docs/readmes/proposals/p025_Converged_GTP_Tunnel_Setup.md]: Yes, Sorry for typo

Comment (prabinakpattnaik) [reply docs/readmes/proposals/p025_Converged_GTP_Tunnel_Setup.md]: I will fix it now

**Review by uri200 — COMMENTED** [review]
i remember that the idea of calling directly ovs from mme was to reduce the amount of messages and avoid converting sessiónd as just a pass through.

Can you describe the new call flow? Is this change saving any messages or i reodicing even more?  Does this change implies session on sessiond modified it's state?

**Comment by uri200** [comment]
Please can you add a diagram with the resulting callflows to see how it will look like? 

You can use mermaid for that so it is easy to edit

https://mermaid-js.github.io/mermaid/#/./sequenceDiagram

**Comment by prabinakpattnaik** [comment]
@uri200, I have attached four call flows diagram for gets more details about gRPC messages. For GTP tunnel creation and Deletion, we can use existing gRPC messages. For GTP Data Notification and Idle Mode session, need to add new gRPC methods.

``` GTP Tunnel Creation ```

GTP_tunnel_Creation

``` GTP Tunnel Deletion ```

GTP_tunnel_Deletion

``` GTP Idle Mode and Paging Notification ```

Idle_Session_Paging_Notification

``` GTP Data Notification ```

Notification_GTP_flows

**Comment by uri200** [comment]
Thanks @prabinakpattnaik . That really helps. Can you add the call flows to the proposal (i think they are just on the comments)

The changes make sense. I think the idea of MME - OVS communication was to reduce the amount of messages and to avoid sessiond-pipelined become just a transit for a message.  

For Create/Terminate events that make sense. They are low frequency events and we don't gain much contacting OVS from MME directly. We still have to contact sessiond and pipelined to create and terminate session.  

But for high frequency events like like `GTP Data Notification`  we go from 1 message to 3 and we have to now go to 3 different systems just to send a message to a service that is sitting next to MME.

Would it make sense to leave `GTP Data Notification` as it is since it looks like it doesn't need any specific information about the session?

**Comment by ekowtaylor** [comment]
@prabinakpattnaik are you still working on this? How can I help?

**Comment by ekowtaylor** [comment]
@prabinakpattnaik do you still plan to move forward with this change? If yes then please reopen and rebase.

**Comment by prabinakpattnaik** [comment]
Sorry for late response. This is on going task. I will update soon.

**Review by Neudrino — COMMENTED** [review]
The graphics look very strange. One should probably use [mermaid diagrams](https://github.com/magma/magma/wiki/Contributing-Documentation#diagrams-within-the-documentation) for this purpose.

**Pull request closed without merging**